src/lib.rs

```diff
@@ -0,0 +1,7 @@
+pub mod rest;
+pub mod websocket;
+
+pub trait OKEx : Sized {
+    fn new() -> Self;
```

Comment:
This isn't a great pattern to do, structs implementing this trait may want their own implementation of the `new` method (with different arguments for example). Maybe strategy pattern would be nicer for this?

Reply:
I think the larger point is that traits shouldn't be used like inheritance in other languages (and particularly shouldn't be used as a replacement to Rust's missing struct inheritance). It's obviously up to opinion but composition over inheritance seems to suit Rust in most situations.